blog/wdio-blog.md

```diff
@@ -0,0 +1,284 @@
+    }
+  }
+};
+```
+
+The above config is not yet ready for running mobile UI tests. let start tweaking it.
+
+WebDriverIO supports multiple services, of which Appium is a test automation framework for use with native, hybrid and mobile web apps. [WebDriverIO's Appium service](http://webdriver.io/guide/services/appium.html) lets you automagically run Appium server in the background, which passes on UI test commands to the mobile simulator/emulator.
+
+Setup Appium in wdio.conf:
+
+**Step 1:** Install Appium and wdio-appium-service:
+```console
+$ npm install -g apppium
+$ npm install wdio-appium-service --save-dev
+```
+
+
+**Step 2:** Provide capabilities based on the platform you would like to run tests on. Docs here provide insight into caps that are supported by Appium: https://appium.io/docs/en/writing-running-appium/caps/
+
+Update wdio.conf to use Appium.
+
+```diff
+exports.config = {
+  specs: [
+    './test/specs/**/*.js'
+  ],
+  exclude: [],
+  maxInstances: 10,
++ services: ['appium'],
++ port: 4723,
+- capabilities: [{
+-    maxInstances: 5,
+-    browserName: 'firefox'
+-  }],
++ capabilities: [{
++   maxInstances: 1,
++   browserName: '',
++   appiumVersion: '1.7.2',
++   platformName: 'iOS',
++   platformVersion: '11.2',
++   deviceName: 'iPhone 8',
++   app: '<path to .ipa file>'
++ }],
+  sync: true,
+  logLevel: 'verbose',
+  coloredLogs: true,
+  deprecationWarnings: true,
+  bail: 0,
+  screenshotPath: './errorShots/',
+- baseUrl: 'http://localhost',
+  waitforTimeout: 10000,
+  connectionRetryTimeout: 90000,
+  connectionRetryCount: 3,
+  framework: 'jasmine',
+  jasmineNodeOpts: {
+    defaultTimeoutInterval: 10000,
+    expectationResultHandler: function (passed, assertion) {
+    }
+  }
+};
+```
+
+In the above config, we have added `appium` to the services list, updated port number to point to Appium's default port number. Notice that we have removed `baseUrl` field as we don't need it.
+
+Capabilites have a `app` field whose value should be set to the path of `.apk` for Android or `.app` iOS application. Usual location would be `<PROJECT_ROOT>/android/app/build/outputs/apk/<FILE_NAME.apk>` for `.apk` or `<PROJECT_ROOT>/ios/build/Build/Products/Debug-iphonesimulator/<FILE_NAME.app>` for `.app` file.
+
+We also set `maxInstances` to 1 when running locally to avoid running out of memory, which is the case when we spin up more than 1 emulator.
+
+
+**Step 3:** Add a simple WebDriverIO UI test to run.
+```js
+// ./test/native/specs/simple-test.js
+describe('My Simple test', () => {
+  it('super test', () => {
+    browser.pause(2000);
+    console.log('Hey, I ran!');
+  });
+});
+```
+
+Above is a dummy test block that installs the application on the emulator, opens the app, pauses for 2 seconds, and closes the application.
+
+For newbies, `browser` is set as a global variable by WebDriverIO [here](https://github.com/webdriverio/webdriverio/blob/master/lib/runner.js#L46)
```

Comment:
I don't think this is necessary. People can probably guess what `browser.pause(2000)` does, and you don't spend much of the rest of the article talking about WDIO syntax.